### DISCUSSION TITLE
Redirect to nextjs error page (_404.tsx) on middleware

### DISCUSSION THREAD

**Discussion by yongsk0066** [vercel/next.js#30682] [Help]
Currently i'm upgrading next.js to v12. and trying to redirect user to 404 on middleware.
is there a way to trigger _404.tsx on nextjs 12 middleware ?

**Comment by merodiro** [comment]
I found that `return NextResponse.rewrite("/404")` works. but I'm not sure if this is the correct way

**Reply by yongsk0066** [reply]
thx! I think that is the best way until now

**Reply by juliankoehn** [reply]
doing it similar

```ts
const url = req.nextUrl;

url.pathname = `/404`;
return NextResponse.rewrite(url);
```

**Reply by mismosmi** [reply]
For completeness' sake:
> doing it similar
> 
> ```ts
> const url = req.nextUrl.clone();
> 
> url.pathname = `/404`;
> return NextResponse.rewrite(url);
> ```

**Reply by hc0503** [reply]
> 

Thanks, that works well.
